cpp/src/fil/internal.cuh

```diff
@@ -406,11 +410,19 @@ struct cat_sets_owner {
     };
   }
 
-  void initialize_from_bit_pool_sizes()
+  void consume_counters(const std::vector<cat_feature_counters>& counters)
```

Comment:
it will not be refactored that way, because `max_matching` is used during inference, whereas `n_nodes` is not, and I am against any extra overhead (it will be an overhead).